### PR TITLE
Do not restart ovn-controller if encap IP doesn't change

### PR DESCRIPTION
**Pull request by trozet**

### CODE REVIEW

**Comment by trozet** [comment]
Looks like it is working as expected. ovn-controller was exiting a bunch of times previously:

```
ovn-control-plane/ovn-kubernetes/ovnkube.log:I0627 22:37:02.020612    1670 node_ip_handler_linux.go:212] Node primary address changed to 172.18.0.3. Updating OVN encap IP.
ovn-control-plane/ovn-kubernetes/ovnkube.log:I0627 22:37:02.037852    1670 node_ip_handler_linux.go:343] Will not update encap IP, value: 172.18.0.3 is the already configured
ovn-control-plane/ovn-kubernetes/ovnkube.log:I0627 23:03:03.115881   16698 node_ip_handler_linux.go:212] Node primary address changed to 172.18.0.3. Updating OVN encap IP.
ovn-control-plane/ovn-kubernetes/ovnkube.log:I0627 23:03:03.122365   16698 node_ip_handler_linux.go:343] Will not update encap IP, value: 172.18.0.3 is the already configured
ovn-control-plane/ovn-kubernetes/ovnkube.log:I0627 23:18:08.381694   27220 node_ip_handler_linux.go:212] Node primary address changed to 172.18.0.3. Updating OVN encap IP.
ovn-control-plane/ovn-kubernetes/ovnkube.log:I0627 23:18:08.389655   27220 node_ip_handler_linux.go:343] Will not update encap IP, value: 172.18.0.3 is the already configured
ovn-control-plane/ovn-kubernetes/ovnkube.log:I0627 23:19:26.638221   28746 node_ip_handler_linux.go:212] Node primary address changed to 172.18.0.3. Updating OVN encap IP.
ovn-control-plane/ovn-kubernetes/ovnkube.log:I0627 23:19:26.644217   28746 node_ip_handler_linux.go:343] Will not update encap IP, value: 172.18.0.3 is the already configured

```

File: go-controller/pkg/node/node_ip_handler_linux.go [hunk 1]
```diff
@@ -327,13 +328,31 @@ func (c *addressManager) nodePrimaryAddrChanged() (bool, error) {
...
+		"get",
+		"Open_vSwitch",
+		".",
+		"external_ids:ovn-encap-ip",
+	}
+	encapIP, stderr, err := util.RunOVSVsctl(checkCmd...)
+	if err != nil {
+		klog.Warningf("Unable to retrieve configured ovn-encap-ip from OVS: %v, %q", err, stderr)
+	} else {
+		encapIP = strings.TrimSuffix(encapIP, "\n")
```
Comment (tssurya) [inline go-controller/pkg/node/node_ip_handler_linux.go:341]: so the issue was that this was acting up on any update even if the changedIP was already configured? That's a bit weird:
```
Node primary address changed to 10.0.141.44. Updating OVN encap IP.
```
I don't get why if the IP .44 was already configured we were trying to do another update for it? Seems like it needs a fix in the higher up layers of "what we consider as a nodeIP update?"

**Review by jcaamano — COMMENTED** [review]
This one is strange, why would ovn-controller restart even if the IP did not change regardless of us setting it or not? I would add a comment being explicit on this so that no one is tempted to remove this change in the future.

File: go-controller/pkg/node/node_ip_handler_linux.go (hunk 1 above)
Comment (ricky-rav) [reply go-controller/pkg/node/node_ip_handler_linux.go:341]: uhm... so `nodePrimaryAddrChanged` returns a wrong value :-/

**Comment by trozet** [comment]
@jcaamano because ovn-controller does not allow for dynamic update of encap IP. I guess @dceara can provide more details with why and we can add a comment

File: go-controller/pkg/node/node_ip_handler_linux.go (hunk 1 above)
Comment (trozet) [reply go-controller/pkg/node/node_ip_handler_linux.go:341]: I agree we should investigate why that happens, but I think it is orthogonal to this change. We should check the OVS config before we do something so drastic as to restart ovn-controller.

Comment (trozet) [reply go-controller/pkg/node/node_ip_handler_linux.go:341]: I filed https://issues.redhat.com/browse/OCPBUGS-15538 to track that issue.

**Pull request merged**